src/app/rpmostree-builtin-status.c

```diff
@@ -198,29 +196,15 @@ status_generic (RPMOSTreeSysroot *sysroot_proxy,
       have_any_live_overlay = have_any_live_overlay || have_live_changes;
     }
 
-  if (txn && txn_path)
+  glnx_unref_object RPMOSTreeTransaction *txn_proxy = NULL;
+  if (!rpmostree_transaction_connect_active (sysroot_proxy, NULL, &txn_proxy,
+                                             cancellable, error))
+    return FALSE;
+
+  if (txn_proxy)
     {
-      const char *method, *sender, *path;
-      g_variant_get (txn, "(&s&s&s)", &method, &sender, &path);
-
-      /* Things currently could race here if the transaction completes after we get the
-       * path.  For now, just ignore errors.  TODO: A more correct fix would involve a loop
-       * on watching the path property, trying a connection, and re-reading the value
-       * and only erroring out if the property hasn't changed.
-       */
-      /* gdbus-codegen started generating autocleanups from 2.50 */
-      glnx_unref_object RPMOSTreeTransactionProxy *txn_proxy =
-        (RPMOSTreeTransactionProxy*)rpmostree_transaction_connect (txn_path, NULL, NULL);
-      if (txn_proxy)
-        {
-          const char *title = rpmostree_transaction_get_title ((RPMOSTreeTransaction*)txn_proxy);
-          g_print ("State: transaction: %s\n", title);
-        }
-      /* Print the address if verbose, *or* if we somehow failed to get
-       * the txn, so we aren't masking errors.
-       */
-      if (opt_verbose || (path && !txn_proxy))
-        g_print ("TransactionAddress: %s %s %s\n", method, sender, path);
+      const char *title = rpmostree_transaction_get_title (txn_proxy);
+      g_print ("State: transaction: %s\n", title);
```

Comment:
Hey, do we also want to care about the case for opt_verbose like we did earlier?

Reply:
It's a good question.  If I remember correctly, the idea I had there was that with `--verbose` it would help debug if the two data (transaction state, transaction active path) were different.  But now that we have a more reliable way to find the active txn, I don't think it's worth keeping.

A user/admin can always directly do e.g.:

> busctl introspect org.projectatomic.rpmostree1 /org/projectatomic/rpmostree1/Sysroot